docs/docs/examples/mini-examples/dynamic-vs-parallel.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Dynamic outputs vs Python parallelism
+description: Comparing Dagster's dynamic outputs with regular Python parallelism for concurrent processing.
+last_update:
+  author: Dennis Hume
+sidebar_custom_props:
+  logo: images/dagster-primary-mark.svg
+---
+
+In this example, we'll explore two different approaches to achieving parallelism in Dagster pipelines. When you have computationally expensive operations that can benefit from parallel processing, you can choose between Dagster's built-in dynamic outputs or standard Python parallelism techniques. Each approach has distinct trade-offs in terms of observability, performance, and resource consumption.
+
+### Problem: Parallel processing of multiple items
+
+Imagine you need to process a collection of items (in our example, letters a-z), where each item requires some computation time (simulated with a sleep operation). Without parallelism, processing each item sequentially would take much longer than processing them concurrently.
+
+The key question is: should you use Dagster's dynamic outputs to create separate op executions for each item, or should you handle the parallelism within a single op using Python's multiprocessing capabilities?
+
+### Solution 1: Dagster dynamic outputs
+
+Dynamic outputs allow you to create separate op executions for each item. In this example, each letter (a-z) gets its own op execution. This approach provides maximum observability and leverages Dagster's built-in retry mechanisms, but comes with additional overhead for each item.
+
+<CodeExample
+  path="docs_projects/project_mini/src/project_mini/defs/dynamic_vs_parallel/dynamic_outputs.py"
+  language="python"
+  title="src/project_mini/defs/dynamic_vs_parallel/dynamic_outputs.py"
+/>
+
+|                    | **Dynamic Outputs Approach**                         |
+| ------------------ | ---------------------------------------------------- |
+| **Execution Time** | ~26 seconds (1 second per letter + Dagster overhead) |
+| **Observability**  | Full visibility into each letter's execution         |
+| **Retry Logic**    | Automatic retry for individual letters               |
+| **Resource Usage** | Each letter consumes a credit in Dagster Cloud       |
+| **Complexity**     | Simple dynamic pattern setup                         |
+
+### Solution 2: Python parallelism
+
+The Python parallelism approach uses multiprocessing within a single op to process all letters concurrently. This provides better performance by utilizing multiple CPU cores, but requires manual implementation of error handling and retry logic.
```

Comment:
Would be helpful to clarify if "better performance" means "better than Dagster dynamic outputs" or "better than not using Python parallelism"